src/query/expression/src/aggregate/group_hash.rs

```diff
@@ -223,8 +225,13 @@ where I: Index
         Ok(())
     }
 
-    fn visit_number<T: Number>(&mut self, buffer: Buffer<T>) -> Result<()> {
-        self.visit_indices(|i| T::upcast_scalar(buffer[i.to_usize()]).agg_hash())
+    fn visit_any_number(&mut self, column: crate::types::NumberColumn) -> Result<()> {
```

Comment:
visit_any_number and visit_any_decimal seems to be useless.

we can directly use `visit_number ` and `visit_decimal ` in group_hash, it would be much easier.

Reply:
How to implement AggHash for any T: Number?